src/autocomplete/databases/postgresql/postgresql-extract-statements.ts

```diff
@@ -0,0 +1,102 @@
+import {Token, tokenize} from '../../shared/tokenize';
+import {postgreSqlAutocompleteData} from './postgresql-autocomplete';
+import {PostgreSqlLexer} from './generated/PostgreSqlLexer';
+import {CursorPosition} from '../../shared';
+
+export interface PostgreSqlStatement {
+    startLine: number;
+    endLine: number;
+    startColumn: number;
+    endColumn: number;
+    statement: string;
+    hasCursor?: boolean;
+}
+
+export function extractPostgreSqlStatementsFromQuery(
+    query: string,
+    cursorPosition?: CursorPosition,
+): PostgreSqlStatement[] {
+    const {tokens} = tokenize(
+        postgreSqlAutocompleteData.Lexer,
+        PostgreSqlLexer.symbolicNames,
+        postgreSqlAutocompleteData.tokenDictionary.SPACE,
+        query,
+    );
+
+    let startLine = 0;
+    let startColumn = 0;
+    let statementStartIndex = 0;
+    let pendingNewStatement = true;
```

Comment:
I don't like this naming. Let's rename it to `newStatementNeeded`

Reply:
I think it's incorrect wording, this variable actually means that now we may process new statement if it exist, `needed` for me means that we should have next statement.

I think we can reverse meaning of this variable and name it as `processingNewStatement`